### PR TITLE
Tune pom

### PR DESCRIPTION
**Pull request by pan3793**
### What changes in this PR?

- Tune `pom.xml`, pin versions and exclude dependencies in `<dependencyManagement/>` block, but set `scope` in specific `<dependency/>` block.
- Exclude lots of unused transitive dependencies.

dist jars diff
```
before                                         after
---                                            ---
apacheds-i18n-2.0.0-M15.jar                    apacheds-i18n-2.0.0-M15.jar
apacheds-kerberos-codec-2.0.0-M15.jar          apacheds-kerberos-codec-2.0.0-M15.jar
api-asn1-api-1.0.0-M20.jar                     api-asn1-api-1.0.0-M20.jar
api-util-1.0.0-M20.jar                         api-util-1.0.0-M20.jar
commons-beanutils-1.7.0.jar                    commons-beanutils-1.7.0.jar
commons-beanutils-core-1.8.0.jar               commons-beanutils-core-1.8.0.jar
                                          >    commons-codec-1.4.jar
commons-collections-3.2.2.jar                  commons-collections-3.2.2.jar
commons-configuration-1.6.jar                  commons-configuration-1.6.jar
commons-digester-1.8.jar                       commons-digester-1.8.jar
                                          >    commons-io-2.4.jar
commons-lang-2.6.jar                           commons-lang-2.6.jar
commons-logging-1.1.3.jar                 <
commons-math-2.2.jar                           commons-math-2.2.jar
curator-client-2.7.1.jar                       curator-client-2.7.1.jar
curator-framework-2.7.1.jar                    curator-framework-2.7.1.jar
curator-recipes-2.7.1.jar                      curator-recipes-2.7.1.jar
curator-test-2.7.1.jar                         curator-test-2.7.1.jar
guava-16.0.1.jar                          |    guava-11.0.2.jar
hadoop-annotations-2.7.4.jar                   adoop-annotations-2.7.4.jar
hadoop-auth-2.7.4.jar                          hadoop-auth-2.7.4.jar
hadoop-common-2.7.4.jar                        hadoop-common-2.7.4.jar
hive-service-rpc-2.3.7.jar                     hive-service-rpc-2.3.7.jar
httpclient-4.2.5.jar                      |    httpclient-4.4.1.jar
httpcore-4.4.1.jar                             httpcore-4.4.1.jar
javassist-3.18.1-GA.jar                        javassist-3.18.1-GA.jar
                                          >    jcl-over-slf4j-1.7.30.jar
kyuubi-common-1.0.1.jar                   |    kyuubi-common-1.1.0-SNAPSHOT.jar
kyuubi-ha-1.0.1.jar                       |    kyuubi-ha-1.1.0-SNAPSHOT.jar
kyuubi-main-1.0.1.jar                     |    kyuubi-main-1.1.0-SNAPSHOT.jar
libfb303-0.9.3.jar                             libfb303-0.9.3.jar
libthrift-0.9.3.jar                            libthrift-0.9.3.jar
log4j-1.2.17.jar                               log4j-1.2.17.jar
netty-3.7.0.Final.jar                          netty-3.7.0.Final.jar
scala-library-2.12.10.jar                      scala-library-2.12.10.jar
slf4j-api-1.7.30.jar                           slf4j-api-1.7.30.jar
slf4j-log4j12-1.7.30.jar                       slf4j-log4j12-1.7.30.jar
zookeeper-3.4.6.jar                            zookeeper-3.4.6.jar
```

### CODE REVIEW

**Comment by yaooqinn** [comment]
the google maven center is for CI like Travis and Github actions to reduce flakiness

**Comment by pan3793** [comment]
> the google maven center is for CI like Travis and Github actions to reduce flakiness

Got it. I will revert this change.

**Comment by pan3793** [comment]
It's ready for review, ping @yaooqinn

File: pom.xml [hunk 1]
```diff
@@ -25,13 +25,13 @@
...
-        <module>kyuubi-ha</module>
+        <module>dev/kyuubi-codecov</module>
         <module>externals/kyuubi-download</module>
         <module>externals/kyuubi-spark-sql-engine</module>
-        <module>kyuubi-main</module>
         <module>kyuubi-assembly</module>
-        <module>dev/kyuubi-codecov</module>
+        <module>kyuubi-common</module>
+        <module>kyuubi-ha</module>
+        <module>kyuubi-main</module>
```
Comment (yaooqinn) [inline pom.xml:34]: why shall we reorder these, the previous one is intentional

File: pom.xml [hunk 2]
```diff
@@ -160,18 +168,213 @@
...
+                <artifactId>kyuubi-ha</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
+            <dependency>
+                <groupId>org.apache.kyuubi</groupId>
+                <artifactId>kyuubi-main</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+
```
Comment (yaooqinn) [inline pom.xml:226]: Never seen a project manage its own subproject like this, what's the purpose here?

File: pom.xml (hunk 1 above)
Comment (pan3793) [reply pom.xml:34]: Just alphabetically. What is the previous sorting logic?

File: pom.xml (hunk 2 above)
Comment (pan3793) [reply pom.xml]: Just another style, which presto used. Yes, it's unusual in apache project. Reverted it.

File: pom.xml (hunk 1 above)
Comment (yaooqinn) [reply pom.xml:34]: the expected build order

Comment (pan3793) [reply pom.xml:34]: Now the dependence between modules declared in specific module's `pom.xml` and managed by maven.

**Comment by yaooqinn** [comment]
I am going to merge this to master for better dev management, thanks very much for your contribution

**Comment by pan3793** [comment]
@yaooqinn Thanks for review.

**Review by yaooqinn — APPROVED** [review]

**Pull request closed without merging**